### ISSUE TITLE
Offer a local data migration/backup/data portability solution built into Fenix

### ISSUE THREAD

**Issue by ghost** [mozilla-mobile/fenix#12957]
This is something that I asked/suggested in #12380. And since _nobody_ bothered to branch that feedback out from that issue. I'm creating a new issue asking the same. Guess all the "[promises to welcome my feedback during this transition](https://github.com/mozilla-mobile/fenix/issues/12228#issuecomment-655258324)" were all probably false! 

Currently, Fenix doesn't offer a local data migration/backup/portability tool. Making it harder for users who don't want to create a Firefox account and/or uncomfortable doing so to easily backup their data. And following that condition this makes it harder to hop/migrate between different versions of Firefox (namely [Firefox for Android](https://play.google.com/store/apps/details?id=org.mozilla.firefox), [Firefox Beta](https://play.google.com/store/apps/details?id=org.mozilla.firefox_beta), [Firefox Nightly](https://play.google.com/store/apps/details?id=org.mozilla.fenix)). This will also make life easier for those users who test your Nightly and Beta versions for you Mozilla. Apart from that it'll also offer data portability which is something I think Mozilla should care if it already hasn't.

Just like there's a feature request for exporting bookmarks #417 into an HTML file, you could make the data backup/portability solution for each data type like for Collections (I'm going to break this into a new issue), logins (again will break our into a new issue), Top Sites (will break out into a new issue), Customised browser settings that could also include installed add-ons names and links to the AMO (breaking out into a new issue), and anything that you think would make this data backup/portability/migration solution more useful. These will also help if the users don't like Fenix and want to move to another browser, for example. :wink: 

And you can offer a solution that could include everything that Firefox account's Sync feature offers into a single data backup file that users can use it to restore possibly in a few clicks in Fenix (Please branch out this into seperate issues yourself if you think it needs those, you can at least do that, right?).

- - -
Tl;dr:

* Offer something like a backup file that could restore all of Fenix's data as it was before re-installing it. This will act as a migration/backup tool that'll bypass needing to create a Firefox account.
* Offer a data portability solution to possibly to each data type that user can export it into some sort of file format widely available and accessible. That they can easily access it and/or restore it into Fenix if they want. Could be possibly really useful for and possibly include these;
  * Collections (I have seen users ask this,  I think this is something probably must be offered). #12960.

* logins stored (could include passwords and usernames that are stored). #12961.
* Top Sites stored -- [Kiwi browser](https://play.google.com/store/apps/details?id=com.kiwibrowser.browser) offers this and exports the Top Sites themselves and their position too into a `.json` file so that when users restore that file it becomes just like as the Top Sites were before uninstalling the browser, perhaps you should observe how others are doing things so that you can improve your Fenix for users! #12958.
* Customised Fenix settings. #12962.
* I'm not sure but it could also include;

* History -- don't know if they'll want this data as portable but you could do your research.

* Names of installed add-ons and links to them -- again do your research.

Well, that was not that much of a Tl;dr I guess... :wink:

- - -
You might say who needs these backup/migration/portability solution. If you're asking that question to yourself then please I request you to browse Reddit for a while where your users are discussing all of this or think about it rather than forcing them to come here on GitHub to file issues. Mozilla should run behind the users to get their feedback not the other way around! Thank you! :grinning:

And by offering this local data backup/migration/portability solution outside the  Firefox account's sync. Mozilla will provide **_choice_** and **_control_** of  the users' data in the hands of the users as they always promise
In their policy and [Mozilla Manifesto Principles](https://www.mozilla.org/en-US/about/manifesto/). And also make _users'_ data more accessible to _them_ and not in some cloud solution (by keeping their data in the Firefox account, you are kind of forcing them to use the Firefox browser to access their data. This must not be like this, the users should decide where their data is stored and make it more accessible and portable. So they can access their data without needing to install Firefox browser everytime. The users should have a complete control over their data).

For me, the browser is like a Platform just like Android, iOS, Windows, Linux, Mac OS, etc.; for the Web. 

- - -
I'm telling you again, please branch out the individual issues if needed or at least help me in doing so. Don't close this issue just as you did with the previous one. I'd be very thankful and grateful to you! I will try to branch out as many as I can. This particular issue could act as a hub for all the issues surrounding it.

P.S. And to anyone who's wondering about that deleted account then let me tell you that it was mine. :grinning: 

It might take some time before I file further issues so wait...!

If anybody has more ideas on this concept then please let me know, I'll file them for you. Thanks. :grinning:

- - -
**Update:** The issues I had filed were closed as duplicates of this issue #12957. For reference, these are the issues I had created: #12958, #12960, #12961, #12962. Thanks. 

* Just to be clear, I'm suggesting 2 solutions here;
  * One would offer a file that users can take the backup into and restore it into another Firefox instance, or restore into a re-installed Firefox instance.
  * Another one would to possibly make each data type more accessible and portable by making each data type that can be exported into a widely available and accessible data file format so they can easily access their data whenever and however way they want. This would also provide some flexibility to the above backup solution I'm proposing.

Thanks for listening to me. Bye. :grinning:

**Comment by kbrosnan** [comment]
Closing as a duplicate of #12957. Until there is a plan to offer export the main issue is enough to track the request. The issue can be reopened if that is prioritized.

**Comment by mhoye** [comment]
Hi, @ShatteredPixel - my name's Mike Hoye, and I'm Mozilla's Engineering community manager. 

About "... all the "promises to welcome my feedback during this transition" were all probably false!", and a selection of other comments you've made to that effect: as much as we value community participation, this is both the public face of our project and our professional working environment, and we're not going going tolerate this ongoing, abrasive, passive aggressive commentary in our issue tracker. 

Like all Mozilla forums, we expect participants in our Github repositories to adhere to Mozilla's community participation guidelines:

https://www.mozilla.org/en-US/about/governance/policies/participation/

As well, participants in all our issue trackers - Bugzilla, Github, and others - should read our Bugzilla Etiquette Guidelines:

https://bugzilla.mozilla.org/page.cgi?id=etiquette.html

... and I strongly encourage you to take them to heart if you intend to continue participating in our development processes.

Thank you.

**Comment by ShatteredPixel** [comment]
> Hi, @ShatteredPixel - my name's Mike Hoye, and I'm Mozilla's Engineering community manager. 
> 
> About "... all the "promises to welcome my feedback during this transition" were all probably false!", and a selection of other comments you've made to that effect: as much as we value community participation, this is both the public face of our project and our professional working environment, and we're not going going tolerate this ongoing, abrasive, passive aggressive commentary in our issue tracker. 
> 
> Like all Mozilla forums, we expect participants in our Github repositories to adhere to Mozilla's community participation guidelines:
> 
> https://www.mozilla.org/en-US/about/governance/policies/participation/
> 
> As well, participants in all our issue trackers - Bugzilla, Github, and others - should read our Bugzilla Etiquette Guidelines:
> 
> https://bugzilla.mozilla.org/page.cgi?id=etiquette.html
> 
> ... and I strongly encourage you to take them to heart if you intend to continue participating in our development processes.
> 
> Thank you.

Thanks for informing me. Now I can freely delete my GitHub account without any regrets. I thought I was helping you people in improving Firefox. That's the only reason I'm here. If I'm causing trouble then there's no reason for me to be here. I did delete my GitHub account but came back to help with at least the bugs I have reported just because I thought the developers might need more information and it's irresponsible to disappear in the middle of nowhere. 


It seems I might have heart lot of people's feeling. That was not my intention at all.



I hope you never see a person like me ever again here. I'm such a bad person, I never realised it. Thanks for realising me that.


I apolozige. And goodbye...

**Comment by m-p-3** [comment]
To be honest, I can somewhat understand the irritation and confusion when the release of Fenix to stable felt rushed to many users who lost their current setup and extensions from Fennec without any clear indication this was going to happen.

**Comment by liuche** [comment]
Duping to #2080
